src/screens/GroupSetting/hooks/useGroupSetting.js

```diff
@@ -2,28 +2,41 @@ import React from 'react';
 import SimpleToast from 'react-native-simple-toast';
 import {launchImageLibrary} from 'react-native-image-picker';
 
-import StringConstant from '../../../utils/string/StringConstant';
+import {Linking} from 'react-native';
+import {useNavigation} from '@react-navigation/core';
+import {useRecoilState} from 'recoil';
 import {Context} from '../../../context';
 import {getChatName} from '../../../utils/string/StringUtils';
 import {requestExternalStoragePermission} from '../../../utils/permission';
-import {uploadFile} from '../../../service/file';
+import SignedMessageRepo from '../../../service/repo/signedMessageRepo';
+import ChannelList from '../../../database/schema/ChannelListSchema';
+import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+import {CHANNEL_GROUP} from '../../../hooks/core/constant';
+import StringConstant from '../../../utils/string/StringConstant';
+import {chatAtom} from '../../../hooks/core/chat/useChatUtilsHook';
+import useUploadImage from '../../../hooks/useUploadImage';
 
-const useGroupSetting = ({navigation, route}) => {
+const useGroupSetting = ({route}) => {
   const [groupChatState] = React.useContext(Context).groupChat;
-
+  const navigation = useNavigation();
   const [channelState] = React.useContext(Context).channel;
   const [profile] = React.useContext(Context).profile;
   const [participants, setParticipants] = React.useState([]);
   const {channel} = channelState;
   const [groupName, setGroupName] = React.useState(
     getChatName(route.params.name, profile.myProfile.username) || 'Set Group Name'
   );
+  const {localDb, refresh} = useLocalDatabaseHook();
+  const [loadingUpdate, setLoadingUpdate] = React.useState(false);
   const [countUser, setCountUser] = React.useState(0);
   const [changeName, setChangeName] = React.useState(false);
   const [changeImage, setChangeImage] = React.useState(false);
   const [base64Profile, setBase64Profile] = React.useState('');
   const [urlImage, setUrlImage] = React.useState(channel?.data?.image);
   const [isLoading, setIsLoading] = React.useState(false);
+  const [chat, setChat] = useRecoilState(chatAtom);
+  const {uploadPhotoImage} = useUploadImage();
+  const {selectedChannel} = chat;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code has introduced a lot of state variables and dependencies. While this is not inherently bad, it does increase the complexity of the component. Make sure that all these states are necessary and consider breaking down the component into smaller ones if it becomes too complex.

```diff
- import {uploadFile} from '../../../service/file';
+ import {Linking} from 'react-native';
+ import {useNavigation} from '@react-navigation/core';
+ import {useRecoilState} from 'recoil';
+ import SignedMessageRepo from '../../../service/repo/signedMessageRepo';
+ import ChannelList from '../../../database/schema/ChannelListSchema';
+ import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+ import {CHANNEL_GROUP} from '../../../hooks/core/constant';
+ import {chatAtom} from '../../../hooks/core/chat/useChatUtilsHook';
+ import useUploadImage from '../../../hooks/useUploadImage';
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->